### PR TITLE
Cleanup/rgb matrix

### LINKED ISSUE #5704: Jellybean and Raindrop animations broke on Ergodox
## Describe the Bug

Animation only shows up on the lift half (the side connected to the IO Expander), 

## System Information

 - Keyboard:  Ergodox EZ Glow
 - Operating system: Windows 10
 - AVR GCC version: 8.3.0
 - QMK Firmware version: 0.6.340

## Additional Context

@XScorpion2 any ideas?

### PR DESCRIPTION
**Pull request by XScorpion2**
## Description

General cleanup pr of rgb matrix. This fixes / removes the need to keymaps to extern the internals of rgb matrix everywhere. This also greatly cleans up all the boiler plate required to add a new effect to rgb matrix as well.

Built on top of PR https://github.com/qmk/qmk_firmware/pull/5786 so merge that PR first.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* Fixes #5704 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c [hunk 1]
```diff
@@ -50,7 +49,6 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 
 
 
-extern led_config_t g_led_config;
 void rgb_matrix_layer_helper (uint8_t red, uint8_t green, uint8_t blue, bool default_layer) {
```
Comment (drashna) [inline keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c]: /cries

Didn't catch this before..... should not have default layer, any more.

**Comment by XScorpion2** [comment]
@drashna fixed the digital rain.

**Comment by drashna** [comment]
Well, it compiles, at least.  It doesn't work right.  But I think that is something that could be put off till later, for now?

**Comment by XScorpion2** [comment]
as much as I hate digital rain, I will fix it tonight

**Comment by drashna** [comment]
yeah, it's one of the more complicated animations. :(

**Comment by XScorpion2** [comment]
@drashna ok, now I know it is fixed!

**Comment by drashna** [comment]
Could the parenthesis notation be added to the doc page, with explicit details why.

**Pull request merged**